compiler/rustc_mir_transform/src/inline.rs

```diff
@@ -279,6 +280,10 @@ impl<'tcx> Inliner<'tcx> {
             }
         }
 
+        if self.tcx.has_attr(callsite.callee.def_id(), sym::rustc_no_mir_inline) {
+            return Err("found #[rustc_no_mir_inline] on callee");
+        }
+
```

Comment:
I think this would be better placed in `check_codegen_attributes` where we check for `#[inline(never)]`